src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -278,15 +279,18 @@ public async Task<bool> TryGetAsync(ArrayBuilder<DiagnosticData> list, Cancellat
                 static bool ShouldIncludeAnalyzer(
                     DiagnosticAnalyzer analyzer,
                     Func<string, bool>? shouldIncludeDiagnostic,
+                    ICodeActionRequestPriorityProvider priorityProvider,
                     DiagnosticIncrementalAnalyzer owner)
                 {
+                    // Skip executing analyzer if its priority does not match the request priority.
+                    if (!priorityProvider.MatchesPriority(analyzer))
```

Comment:
This check was being once per each analysis kind for every analyzer. I moved it up front so it is done only once for every analyzer.